Jdp200203 11

mackowit opened the pull request.

hacker-msl commented on src/main/java/com/kodilla/ecommercee/domain/Group.java:
```diff
@@ -0,0 +1,50 @@
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+
+    @Column(name = "name")
```
e jakiego powodu nadpisujesz nazwe "name" na "name" ?

hacker-msl commented on src/main/java/com/kodilla/ecommercee/domain/Group.java:
```diff
@@ -0,0 +1,50 @@
+import java.util.ArrayList;
+import java.util.List;
+
+@Entity(name = "groups")
```
specjalnie zmieniasz nazwe groups na groups ?

mackowit replied:
probowalem pozostawic samo @Entity ale wyrzuca kompilator błąd, dlatego pozostawiam "name'y" przy @Entity, ale przy @Column usunąłem

hacker-msl commented on src/main/java/com/kodilla/ecommercee/domain/Order.java:
```diff
@@ -0,0 +1,82 @@
+import java.util.ArrayList;
+import java.util.List;
+
+@Entity(name="orders")
```
uwazasz ze zmiana nazwy cos tutaj daje? Samo @Entity by nie wystarczylo?

hacker-msl commented on src/main/java/com/kodilla/ecommercee/domain/Order.java:
```diff
@@ -0,0 +1,82 @@
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "order_id")
```
ja uwazam, ze to bedzie tez dzialalo bez adnotacji @Column albo z nia bez zmiany nazwy. Specjalnie wszedzie zmieniasz nazwy?

mackowit replied:
jasne, usuwam

hacker-msl commented on src/main/java/com/kodilla/ecommercee/domain/Order.java:
```diff
@@ -0,0 +1,82 @@
+    private Long id;
+
+    @NotNull
+    @Column(name = "number")
```
ja uwazam, ze to bedzie tez dzialalo bez adnotacji @Column albo z nia bez zmiany nazwy. Specjalnie wszedzie zmieniasz nazwy?

hacker-msl commented on src/main/java/com/kodilla/ecommercee/domain/Order.java:
```diff
@@ -0,0 +1,82 @@
+    private String number;
+
+    @NotNull
+    @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
```
jeden uzytkownik moze miec tylko jedno zamowienie? czy to na pewno jest OK ?

mackowit replied:
jasne, błąd logiczny :)

hacker-msl commented:
jak to poprawisz to wypchnij zmiany i popros o ponowne review. na razie nie widze Twoich zmian

hacker-msl commented:
ten PR odnosi sie do zadania https://projects.kodilla.com/projects/JDP200203/issues/JDP200203-11?filter=allopenissues ? Jesli tak to powinien zawierac jedynie Encje order a pozostale encje powinny byc po prostu puste.

mackowit commented:
W tym PR powinien się znaleźć jedynie plik encji Order oraz jego repozytorium, zapewne coś źle wypchnąłem skoro w jednym PR połączyły się dwa commity: właściwy oraz ten który wysyłałem chwilę wcześniej JDP200203-13. Może masz jakąś poradę aby uniknąć tego w przyszłości, wydaje mi się, że w sekcji "staged" były tylko właściwe 2 pliki.

mackowit commented:
Pytanie jak można to naprawić, tak żeby ten PR zawierał tylko to co trzeba, Czy można usunąć zbędne pliki, czy zamknąc całego PR i zrobić go od nowa?

hacker-msl commented:
po prostu skasuj te pliki, które mają wylecieć, zrob commit i wypchnij zmiany, powinno być OK. Czy ten PR u Ciebie się buduje? Bo u mnie coś się nie udaje... stworzyłeś listę User ale nie ma w projekcie klasy User... jeśli mamy zrobić merge tego PR to powinien on się budować. Jeśli musisz dodać puste klasy aby PR się budował, zrób to, proszę...

mackowit commented:
czekałem na ustalenia z wideo, dziś wydaje się że dogadaliśmy już szczegóły, więc jutro myślę to dokończyć i spróbować wypchnąć

mackowit commented:
wysłałem poprawioną encję ,mam nadzieję że tym razem dobrze to zrobiłem (mam na myśli wysłanie)

hacker-msl commented on src/main/java/com/kodilla/ecommercee/domain/Order.java:
```diff
@@ -0,0 +1,85 @@
+import java.util.ArrayList;
+import java.util.List;
+
+@Entity(name = "orders")
```
prosiłem, żebyś usunął name = orders
to nie jest nam tutaj potrzebne.

mackowit replied:
nie da rady, w sql słowo order jest zarezerwowane i przy zapytaniach wyrzuca błędy, tak samo group

hacker-msl replied:
ok, masz rację tutaj, nie przemyślałem tego, zostaw :-)

hacker-msl commented on src/main/java/com/kodilla/ecommercee/domain/Order.java:
```diff
@@ -0,0 +1,85 @@
+    @ManyToOne(
+            //cascade = CascadeType.ALL,
+            fetch = FetchType.EAGER)
+    @JoinColumn(name = "user_id")
```
atrybut "name = user_id" nie jest potrzebny w mojej ocenie

hacker-msl replied:
w zasadzie to cała adnotacja @JoinColumn jest niepotrzebna, do wyrzucenia.

hacker-msl commented:
dodaj pustą klasę Cart gdzie dodasz adnotacje @Entity i pole @Id.

mackowit commented:
z tym Carts jest o tyle problem, że w kontruktorze Order wykorzystuję listę produktów i usera z klasy Cart, więc pasowałoby jeszcze dodać właśnie te pola i gettery do nich

mackowit commented:
ale jak to zrobiłem to mi się nie buduje projekt, może będziesz mieć jakiś pomysł dlaczego
com.kodilla.ecommercee.EcommerceeApplicationTests > contextLoads FAILED
    java.lang.IllegalStateException
        Caused by: org.springframework.beans.factory.BeanCreationException
            Caused by: javax.persistence.PersistenceException
                Caused by: org.hibernate.MappingException

com.kodilla.ecommercee.SpringBootJPAIntegrationTest > givenGenericEntityRepository_whenSaveAndRetreiveEntity_thenOK FAILED
    java.lang.IllegalStateException
        Caused by: org.springframework.beans.factory.BeanCreationException
            Caused by: javax.persistence.PersistenceException
                Caused by: org.hibernate.MappingException

klasa Cart wygląda tak:
package com.kodilla.ecommercee.domain;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
public class Cart {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private User user;

    private List<Product> productsList = new ArrayList<>();

    public Cart() {
    }

    public List<Product> getProductsList() {
        return productsList;
    }

    public User getUser() {
        return user;
    }
}

mackowit commented:
Ciężka sprawa, potestowałem to trochę i wydaję mi się,, że żeby to się budowało, trzeba zbudować encję Cart wraz z zależnościami. Jedynym wyjściem pośrednim jest stworzenie zwykłej klasy Cart z dwoma polami i dwoma getterami, ale to pewnie nie wchodzi w rachubę.

hacker-msl commented on src/main/java/com/kodilla/ecommercee/domain/Order.java:
```diff
@@ -0,0 +1,85 @@
+            name = "JOIN_PRODUCTS_ORDERS",
+            joinColumns = @JoinColumn(name = "order_id"),
+            inverseJoinColumns = @JoinColumn(name = "product_id")
+    )
```
bez @JoinTable też u mnie działa.

hacker-msl commented on src/main/java/com/kodilla/ecommercee/domain/Order.java:
```diff
@@ -0,0 +1,85 @@
+        this.user = cart.getUser();
+        this.orderDate = orderDate;
+        this.orderedProductsList = cart.getProductsList();
+    }
```
W jakim celu stworzyłeś dodatkowy konstruktor? Mamy od tego settery a tak stworzony konstruktor powoduje, że musisz w pełni zaimplementować klasę Cart. Wyrzuć ten kod, przynajmniej na razie.

mackowit commented:
zmiany wprowadzone zgodnie z uwagami

hacker-msl commented:
hej, klepnięte. Niestety mamy 2 konflikty, rozwiąż je robiąc merge z mastera do Twojego brancha. 
Conflicting files
src/main/java/com/kodilla/ecommercee/domain/Cart.java
src/main/java/com/kodilla/ecommercee/domain/User.java

mackowit commented:
zrobiłem git pull origin master, ściągnęło mi pliki z mastera i wychodzi na to, że konflikty są w tych szczątkowych klasach User i Cart. Rozumiem, że powinienem teraz podmienić zawartośc moich szczątkowych klas na klasy ściągnięte z mastera i jeszcze raz wszystko wypchnąć do PR?

mackowit commented:
zrobiłem jak napisałem, nie wiem czy dobrze :) ale nie wiem teraz gdzie mogą być konflikty z Group i GroupRepository. te klasy nie są ze sobą powiązane wogóle

hacker-msl commented:
konflikt istnieje ponieważ te pliki usunęliśmy z tego PR. 

mateo@linux-to0r:~/eclipse-workspace/kodilla/project-jdp-2002-03> git merge master
CONFLICT (modify/delete): src/main/java/com/kodilla/ecommercee/repository/GroupRepository.java deleted in HEAD and modified in master. Version master of src/main/java/com/kodilla/ecommercee/repository/GroupRepository.java left in tree.
CONFLICT (modify/delete): src/main/java/com/kodilla/ecommercee/domain/Group.java deleted in HEAD and modified in master. Version master of src/main/java/com/kodilla/ecommercee/domain/Group.java left in tree.
Automatic merge failed; fix conflicts and then commit the result.
mateo@linux-to0r:~/eclipse-workspace/kodilla/project-jdp-2002-03> 

Rozwiąż konflikty w IntelliJ i wypchnij zmiany. Musisz te pliki z mastera umieścić u siebie teraz.

mackowit commented:
Mam te pliki właśnie już z mastera ściągnięte u siebie i nie ma w nich żadnych adnotacji o konfliktach

mackowit commented:
kurczę, próbowałem rozwiązać te konflikty i chyba coś zrobiłem źle, bo wypchnąłęm commitem wszystkie pliki, proszę o poradę jak to teraz rozwiązać, w files changed jest w sumie to co trzeba

mackowit commented:
czy wystarczy jak w swoim lokalnym branchu usunę te pliki git rm <filename> a potem zrobię commit z jednym plikiem np. Order.java? a potem sobie je ponownie ściągne git pull origin master?

mackowit commented:
usunąłem zbędne pliki, chciałem dodać jeszcze user.java ale nie chciał się dodać, więc mniemam, że dlatego że jest już na masterze

hacker-msl commented:
hej, dalej mamy konflikt z 4 plikami. Powiedz czy ogarniesz @mackowit ?

mackowit commented:
szczerze mówiąc to nie wiem co z tym zrobić w tej chwili, jesli to możliwe to proszę o pomoc
po zrobieniu git pull origin master mam taki komunikat:
C:\Users\Samsung\Projects\project-jdp-2002-03>git pull origin master
From https://github.com/hacker-msl/project-jdp-2002-03
 * branch            master     -> FETCH_HEAD
CONFLICT (modify/delete): src/main/java/com/kodilla/ecommercee/service/GroupService.java deleted in HEAD and modified in 39253a365da58f1eace4fdfd8ce5d624b99a71d5. Version 39253a365da58f1eace4fdfd8ce5d624b99a71d5 of src/
main/java/com/kodilla/ecommercee/service/GroupService.java left in tree.
CONFLICT (modify/delete): src/main/java/com/kodilla/ecommercee/service/CartService.java deleted in HEAD and modified in 39253a365da58f1eace4fdfd8ce5d624b99a71d5. Version 39253a365da58f1eace4fdfd8ce5d624b99a71d5 of src/m
ain/java/com/kodilla/ecommercee/service/CartService.java left in tree.
CONFLICT (modify/delete): src/main/java/com/kodilla/ecommercee/repository/GroupRepository.java deleted in HEAD and modified in 39253a365da58f1eace4fdfd8ce5d624b99a71d5. Version 39253a365da58f1eace4fdfd8ce5d624b99a71d5 o
f src/main/java/com/kodilla/ecommercee/repository/GroupRepository.java left in tree at src/main/java/com/kodilla/ecommercee/repository/GroupRepository.java~39253a365da58f1eace4fdfd8ce5d624b99a71d5.
CONFLICT (modify/delete): src/main/java/com/kodilla/ecommercee/domain/UserDto.java deleted in HEAD and modified in 39253a365da58f1eace4fdfd8ce5d624b99a71d5. Version 39253a365da58f1eace4fdfd8ce5d624b99a71d5 of src/main/j
ava/com/kodilla/ecommercee/domain/UserDto.java left in tree.
CONFLICT (modify/delete): src/main/java/com/kodilla/ecommercee/domain/GroupDto.java deleted in HEAD and modified in 39253a365da58f1eace4fdfd8ce5d624b99a71d5. Version 39253a365da58f1eace4fdfd8ce5d624b99a71d5 of src/main/
java/com/kodilla/ecommercee/domain/GroupDto.java left in tree.
CONFLICT (modify/delete): src/main/java/com/kodilla/ecommercee/domain/Group.java deleted in HEAD and modified in 39253a365da58f1eace4fdfd8ce5d624b99a71d5. Version 39253a365da58f1eace4fdfd8ce5d624b99a71d5 of src/main/jav
a/com/kodilla/ecommercee/domain/Group.java left in tree at src/main/java/com/kodilla/ecommercee/domain/Group.java~39253a365da58f1eace4fdfd8ce5d624b99a71d5.
CONFLICT (rename/delete): src/main/java/com/kodilla/ecommercee/GroupController.java deleted in HEAD and renamed to src/main/java/com/kodilla/ecommercee/controller/GroupController.java in 39253a365da58f1eace4fdfd8ce5d624
b99a71d5. Version 39253a365da58f1eace4fdfd8ce5d624b99a71d5 of src/main/java/com/kodilla/ecommercee/controller/GroupController.java left in tree.
CONFLICT (modify/delete): src/main/java/com/kodilla/ecommercee/CartController.java deleted in HEAD and modified in 39253a365da58f1eace4fdfd8ce5d624b99a71d5. Version 39253a365da58f1eace4fdfd8ce5d624b99a71d5 of src/main/j
ava/com/kodilla/ecommercee/CartController.java left in tree.
Automatic merge failed; fix conflicts and then commit the result

po komendzie git status jest tak
Changes to be committed:
        new file:   src/main/java/com/kodilla/ecommercee/controller/UserController.java
        renamed:    src/main/java/com/kodilla/ecommercee/GenericEntity.java -> src/main/java/com/kodilla/ecommercee/domain/GenericEntity.java
        new file:   src/main/java/com/kodilla/ecommercee/domain/Key.java
        modified:   src/main/java/com/kodilla/ecommercee/domain/User.java
        new file:   src/main/java/com/kodilla/ecommercee/exception/GroupNotFoundException.java
        new file:   src/main/java/com/kodilla/ecommercee/exception/UserNotAuthorisedException.java
        new file:   src/main/java/com/kodilla/ecommercee/exception/UserNotFoundException.java
        new file:   src/main/java/com/kodilla/ecommercee/mapper/GroupMapper.java
        new file:   src/main/java/com/kodilla/ecommercee/mapper/UserMapper.java
        renamed:    src/main/java/com/kodilla/ecommercee/GenericEntityRepository.java -> src/main/java/com/kodilla/ecommercee/repository/GenericEntityRepository.java
        new file:   src/main/java/com/kodilla/ecommercee/repository/UserRepository.java
        new file:   src/main/java/com/kodilla/ecommercee/service/UserService.java
        new file:   src/test/java/com/kodilla/ecommercee/controller/GroupControllerTest.java

Unmerged paths:
  (use "git add/rm <file>..." as appropriate to mark resolution)
        deleted by us:   src/main/java/com/kodilla/ecommercee/CartController.java
        added by them:   src/main/java/com/kodilla/ecommercee/controller/GroupController.java
        deleted by us:   src/main/java/com/kodilla/ecommercee/domain/Group.java
        deleted by us:   src/main/java/com/kodilla/ecommercee/domain/GroupDto.java
        deleted by us:   src/main/java/com/kodilla/ecommercee/domain/UserDto.java
        deleted by us:   src/main/java/com/kodilla/ecommercee/repository/GroupRepository.java
        deleted by us:   src/main/java/com/kodilla/ecommercee/service/CartService.java
        deleted by us:   src/main/java/com/kodilla/ecommercee/service/GroupService.java


C:\Users\Samsung\Projects\project-jdp-2002-03>git stash
src/main/java/com/kodilla/ecommercee/CartController.java: needs merge
src/main/java/com/kodilla/ecommercee/controller/GroupController.java: needs merge
src/main/java/com/kodilla/ecommercee/domain/Group.java: needs merge
src/main/java/com/kodilla/ecommercee/domain/GroupDto.java: needs merge
src/main/java/com/kodilla/ecommercee/domain/UserDto.java: needs merge
src/main/java/com/kodilla/ecommercee/repository/GroupRepository.java: needs merge
src/main/java/com/kodilla/ecommercee/service/CartService.java: needs merge
src/main/java/com/kodilla/ecommercee/service/GroupService.java: needs merge

poprzednio przy takim komunikacie usunąłem pliki wskazane w konliktach poprzez git rm <filename> i wypchnąłem commit do PR, czy teraz powinienem zrobić to samo?
ale najpierw musiałbym dodać te pliki z Unmerged paths poprzez git add, potem zrobić git stash a potem chyba znowu git rm i potem commit
a może po prostu trzeba wysłać te nowe pliki do PR i one się potem podmienią w masterze jak zrobisz merge?
będę wdzięczny za pomoc w tym zakresie

hacker-msl commented:
@mackowit rozwiąż te konflikty w IntelliJ. Dlaczego próbujesz to zrobić w konsoli? Dzisiaj na video zespołowym pokażę jak rozwiązywać tego typu problemy.

hacker-msl commented:
not needed as #24 created
